Lib/functools.py

```diff
@@ -423,6 +423,17 @@ def _unwrap_partial(func):
         func = func.func
     return func
 
+def _unwrap_partialmethod(func):
+    prev = None
+    while func is not prev:
+        prev = func
+        while isinstance(getattr(func, "_partialmethod", None), partialmethod):
```

Comment:
Why getattr? What objects have `_partialmethod`?

Reply:
`partialmethod(...).__get__()` can return a closure with `_partialmethod` pointing to the instance. This happens if the wrapped callable is not a descriptor, or if the descriptor returned itself again.

See the [`partialmethod._make_unbound_method()` implementation](https://github.com/python/cpython/blob/98308dbeb110198ebe28bdb7720d3671b3e7f57b/Lib/functools.py#L385-L391).

`getattr()` is used because the attribute is optional, and you'd not want this to break even if the supplied object does have a `_partialmethod` attribute that is not actually a `partialmethod` instance.